lte/gateway/python/magma/monitord/cpe_monitoring.py

```diff
@@ -0,0 +1,100 @@
+"""
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import logging
+from collections import defaultdict
+from datetime import datetime
+from typing import Dict, List
+import ipaddress
+
+import grpc
+from lte.protos.mobilityd_pb2 import IPAddress, SubscriberIPTable
+from lte.protos.mobilityd_pb2_grpc import MobilityServiceStub
+from magma.common.rpc_utils import grpc_async_wrapper
+from magma.common.service_registry import ServiceRegistry
+from magma.magmad.check.network_check.ping import PingCommandResult
+from magma.monitord.icmp_state import ICMPMonitoringResponse
+from orc8r.protos.common_pb2 import Void
+from prometheus_client import Histogram
+from magma.subscriberdb.sid import SIDUtils
+from magma.configuration import load_service_config
+
+SUBSCRIBER_ICMP_LATENCY_MS = Histogram('subscriber_icmp_latency_ms',
+                                  'Reported latency for subscriber '
+                                  'in milliseconds',
+                                  ['imsi'],
+                                  buckets=[50, 100, 200, 500, 1000, 2000])
+
+def _get_addr_from_subscribers(sub) -> str:
+    return str(ipaddress.IPv4Address(
+        sub.ip.address.decode()) if sub.ip.version == 0 else \
+                   ipaddress.IPv6Address(sub.ip.address.decode()))
+
+class CpeMonitoring():
```

Comment:
this could be renamed to `CPEMonitoringModule`

Reply:
Review comment incorporated.